packages/components/pie-cookie-banner/test/helpers/page-object/pie-cookie-banner.page.ts

```diff
@@ -0,0 +1,284 @@
+     *                          has been successfully clicked.
+     */
+    async clickManagePreferencesAction () : Promise<void> {
+        await this.managePrefsLinkLocator.click();
+    }
+
+    /**
+     * Clicks the "Manage Preferences" link within the body.
+     *
+     * @returns {Promise<void>} A Promise that resolves once the "Manage Preferences" link
+     *                          within the body has been successfully clicked.
+     */
+    async clickManagePreferencesLink () : Promise<void> {
+        await this.bodyManagePreferencesLinkLocator.click();
+    }
+
+    /**
+     * Clicks either the "Manage Preferences" action or the "Manage Preferences" link within the body, based on the specified element level.
+     *
+     * @param {Level} elementLevel The level of the element to click, either 'action' or 'body'.
+     * @returns {Promise<void>} A Promise that resolves once the appropriate element has been successfully clicked.
+     */
+    async clickManagePreferencesAll (elementLevel: Level) : Promise<void> {
+        await (elementLevel === 'action' ? this.clickManagePreferencesAction() : this.clickManagePreferencesLink());
+    }
+
+    /**
+     * Clicks the preference toggle associated with the specified preference IDs.
+     *
+     * @param {PreferenceIds} preferenceIds The preference IDs used to locate the preference toggle.
+     * @returns {Promise<void>} A Promise that resolves once the preference toggle has been successfully clicked.
+     */
+    async clickPreferenceToggle (preferenceIds: PreferenceIds) : Promise<void> {
+        const preferenceToggle = this.page.locator(getPreferenceItemSelector(preferenceIds));
+        return preferenceToggle.click();
+    }
+
+    /**
+     * Checks whether the preference toggle associated with the specified preference IDs is checked.
+     *
+     * @param {PreferenceIds} preferenceIds The preference IDs used to locate the preference toggle.
+     * @returns {Promise<boolean>} A Promise that resolves to a boolean value indicating whether the preference toggle is checked (`true`) or not (`false`).
+     */
+    async isPreferenceToggleChecked (preferenceIds: PreferenceIds) : Promise<boolean> {
+        const preferenceToggle = this.page.locator(getPreferenceItemSelector(preferenceIds));
+        return preferenceToggle.isChecked();
+    }
+
+    /**
+     * Checks whether the preference toggle associated with the specified preference IDs is disabled.
+     *
+     * @param {PreferenceIds} preferenceIds The preference IDs used to locate the preference toggle.
+     * @returns {Promise<boolean>} A Promise that resolves to a boolean value indicating whether the preference toggle is disabled (`true`) or not (`false`).
+     */
+    async isPreferenceToggleDisabled (preferenceIds: PreferenceIds) : Promise<boolean> {
+        const preferenceToggle = this.page.locator(getPreferenceItemSelector(preferenceIds));
+        return preferenceToggle.isChecked();
```

Comment:
i might be missing something, but based on the method names, we should check for preferenceToggle.isDisabled() right?

Reply:
Oops yeah, I think you're right! I'll fix now.